valondigital/timesheet
# 🚀 Feature: Paginating All Models

👋 Hello awesome team! I hope you're all doing great! 🎉

I'm excited to share my latest work on paginating all models in our timesheet application. This PR introduces a convenient and efficient way for users to navigate through large datasets without overwhelming the user interface. 📊

### What's New 🆕

- Added pagination functionality to all models in the application, including:
  - User logs
  - Project tasks
  - Employee attendance records

### Features and Changes ⭐

1. **User Logs Pagination:**
   - Users can now easily browse their logs without loading excessive data at once.
   - Improved user experience by dividing logs into manageable chunks.
   - Implemented 'Next' and 'Previous' buttons for seamless navigation.

2. **Project Task Pagination:**
   - Project managers can efficiently manage tasks without performance bottlenecks.
   - Pagination ensures smooth task overview, even for projects with numerous tasks.

3. **Employee Attendance Records Pagination:**
   - HR teams can now review attendance records effortlessly.
   - Large datasets are broken down into pages for better readability.

### How to Test 🧪

1. Pull down this branch and ensure you have the latest updates.
2. Run the application locally.
3. Navigate to relevant sections (logs, tasks, attendance) and observe the new pagination controls.
4. Test navigation between pages and verify that data is loaded correctly.

### Additional Notes 📝

- Pagination logic is implemented using a reusable component for consistent UI across the application.
- The UI gracefully handles edge cases, such as no data to display on a particular page.

### Checklist ✅

- [x] Code has been tested thoroughly.
- [x] Documentation is updated to reflect the pagination feature.
- [x] Unit tests for the pagination component are included.

### Future Improvements 🔮

- We could explore adding additional filter options to further enhance user experience.

I'm thrilled to have contributed to this project and am eager to hear your feedback! Please let me know if there are any changes or improvements you'd like to suggest. 🙌

Thank you for your time and collaboration! 👏👏

Best regards,
Ayo Moses
